### PR TITLE
Define mechanics properties

### LINKED ISSUE #9800: Change get/set methods in vector and mechanics modules to @property
In `sympy/physics/vector` and `sympy/physics/mechanics` most of the classes have lots of getter/setter methods. It would be nice to change the API to be more pythonic by making use of the `@property` decorator. It would be great to do this before a SymPy 1.0 release.

@sympy/mechanics

### PR DESCRIPTION
**Pull request by oliverlee**
This resolves https://github.com/sympy/sympy/issues/9800

### CODE REVIEW

**Comment by moorepants** [comment]
@sahilshekhawat Please review this PR for Oliver so you can get it in before Body and make use of it.

**Comment by oliverlee** [comment]
Do you know why the test is failing?

**Comment by moorepants** [comment]
I often ignore the pypy test failures. I restarted it to see if it will pass, but this can probably be merged regardless. Let's wait and see.

**Comment by moorepants** [comment]
@sahilshekhawat Please review this as we want to get it in before Body so you can change your tests.

**Comment by sahilshekhawat** [comment]
Yeah, I just restart it and most of the time it is passed.

**Comment by sahilshekhawat** [comment]
@moorepants  Definitely.

File: sympy/physics/mechanics/particle.py [hunk 1]
```diff
@@ -42,34 +43,66 @@ def __init__(self, name, point, mass):
...
+                deprecated_since_version="0.7.7", issue=9800).warn()
+        return self.mass
+
+    def set_mass(self, value):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics." +
+                    "Particle.set_mass(self, value)",
+                useinstead="property sympy.physics.mechanics.Particle.mass",
+                deprecated_since_version="0.7.7", issue=9800).warn()
+        self.mass = value
```
Comment (sahilshekhawat) [inline sympy/physics/mechanics/particle.py]: Maybe we should sympify the value here too?

File: sympy/physics/mechanics/particle.py [hunk 2]
```diff
@@ -42,34 +43,66 @@ def __init__(self, name, point, mass):
...
+                useinstead="property sympy.physics.mechanics.Particle.point",
+                deprecated_since_version="0.7.7", issue=9800).warn()
+        return self.point
+
+    def set_point(self, p):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics.Particle." +
+                    "set_point(self, p)",
+                useinstead="property sympy.physics.mechanics.Particle.point",
+                deprecated_since_version="0.7.7", issue=9800).warn()
```
Comment (sahilshekhawat) [inline sympy/physics/mechanics/particle.py]: Please add TypeError here too.

File: sympy/physics/mechanics/particle.py [hunk 3]
```diff
@@ -215,10 +251,17 @@ def potential_energy(self):
...
+        self._pe = sympify(scalar)
+
+    def set_potential_energy(self, scalar):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics." +
+                    "Particle.set_potential_energy(self, scalar)",
+                useinstead="property sympy.physics.mechanics." +
+                    "Particle.potential_energy",
+                deprecated_since_version="0.7.7", issue=9800).warn()
+        self.potential_energy = scalar
```
Comment (sahilshekhawat) [inline sympy/physics/mechanics/particle.py]: Please add ``sympify(scalar)`` here too.

File: sympy/physics/mechanics/rigidbody.py [hunk 4]
```diff
@@ -51,49 +52,103 @@ def __init__(self, name, masscenter, frame, mass, inertia):
...
 
-    frame = property(get_frame, set_frame)
+    def get_frame(self):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics." +
+                    "RigidBody.get_frame(self)",
+                useinstead="property sympy.physics.mechanics." +
+                    "RigidBody.frame",
+                deprecated_since_version="0.7.7", issue=9800).warn()
+        return self.frame
```
Comment (sahilshekhawat) [inline sympy/physics/mechanics/rigidbody.py]: should be ``return self._frame``

File: sympy/physics/mechanics/rigidbody.py [hunk 5]
```diff
@@ -51,49 +52,103 @@ def __init__(self, name, masscenter, frame, mass, inertia):
...
 
-    def get_masscenter(self):
+    def set_frame(self, F):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics." +
+                    "RigidBody.set_frame(self, F)",
+                useinstead="property sympy.physics.mechanics." +
+                    "RigidBody.frame",
+                deprecated_since_version="0.7.7", issue=9800).warn()
+        self.frame = F
```
Comment (sahilshekhawat) [inline sympy/physics/mechanics/rigidbody.py]: Here too, frame should be private.

File: sympy/physics/mechanics/rigidbody.py [hunk 6]
```diff
@@ -51,49 +52,103 @@ def __init__(self, name, masscenter, frame, mass, inertia):
...
 
-    masscenter = property(get_masscenter, set_masscenter)
+    def get_masscenter(self):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics." +
+                    "RigidBody.get_masscenter(self)",
+                useinstead="property sympy.physics.mechanics." +
+                    "RigidBody.masscenter",
+                deprecated_since_version="0.7.7", issue=9800).warn()
+        return self.masscenter
```
Comment (sahilshekhawat) [inline sympy/physics/mechanics/rigidbody.py]: masscenter should be private.

File: sympy/physics/mechanics/rigidbody.py [hunk 7]
```diff
@@ -51,49 +52,103 @@ def __init__(self, name, masscenter, frame, mass, inertia):
...
 
-    def get_mass(self):
+    def set_masscenter(self, p):
+        SymPyDeprecationWarning(
+                feature="Method sympy.physics.mechanics." +
+                    "RigidBody.set_masscenter(self, p)",
+                useinstead="property sympy.physics.mechanics." +
+                    "RigidBody.masscenter",
+                deprecated_since_version="0.7.7", issue=9800).warn()
+        self.masscenter = p
```
Comment (sahilshekhawat) [inline sympy/physics/mechanics/rigidbody.py]: Here too, masscenter should be private.

Comment (sahilshekhawat) [inline sympy/physics/mechanics/rigidbody.py]: Also, please add a TypeError here.

Comment (sahilshekhawat) [inline sympy/physics/mechanics/rigidbody.py]: Got it, sorry, You are calling setter of masscenter. Perfect. its correct.

**Comment by sahilshekhawat** [comment]
Everything looks good. Please change those lines in docs and it is ready to merge. Thank you for this PR.

**Comment by oliverlee** [comment]
done: https://github.com/oliverlee/sympy/commit/ef65a76bfff9aadf8ca357b315ed82ca0c7bae2e

**Comment by sahilshekhawat** [comment]
@moorepants it looks good to go in.

**Comment by sahilshekhawat** [comment]
@oliverlee This is good to be merged from my side.

**Comment by moorepants** [comment]
Tests pas. Merging.

**Comment by sahilshekhawat** [comment]
Thank you Jason.

**Pull request merged**